dev/source/docs/ardupilot-on-linux-kernel-configuration.rst

```diff
@@ -0,0 +1,38 @@
+.. _ardupilot-on-linux-kernel-configuration:
+
+==========================
+Linux Kernel Configuration
+==========================
+
+Real-Time Patches
+=================
+For some builds, ArduPilot needs to run sensors, control loop, and output
+interfaces at high rates, with bounded latency to maintain stable
+operation. For this reason, it is suggested to run a real-time kernel.
+
+Builds like rover, sub, and tracker may be fine without it.
+
+Userspace access to hardware peripherals on 64-bit boards
+=========================================================
+The default kernel configuration for many 64-bit boards blocks
+even the root user from mmap()ing the peripheral device address space,
+allowing only the kernel to access it.
+
+For many boards, ArduPilot will need such access, so CONFIG_STRICT_DEVMEM
+or CONFIG_IO_STRICT_DEVMEM will need to be disabled.
+
+32-bit kernels are unaffected.
+
+Preventing CPU idle states
+==========================
+The kernel command line option cpuidle.off=1 prevents CPU cores from
+entering idle states. This can be useful on CPUs where exiting idle
+states is slow.
+
+Isolating a core for ArduPilot
+==============================
+When running multiple real-time programs on a multi-core CPU, we can isolate a
+CPU from general-purpose scheduling and IRQs and dedicate it to ArduPilot.
+Example parameters to isolate core 3 of a quad-core system from IRQs, scheduling, and RCUs:
```

Comment:
shouldn't we recommand to use cpu0 by default (people will just copy this line) as that the one that generally is doing io and periph access ? So that maximize performance

Reply:
>shouldn't we recommand to use cpu0 by default (people will just copy this line) as that the one that generally is doing io and periph access ? So that maximize performance

It depends on the system. Without an irqbalancer, the boot CPU(often cpu0) will handle IRQs by default. But, this also means that the boot CPU can be busier doing so. Running AP and its IRQs(SPI, UART, I2C) on an isolated core has worked well for me on RPi4, but this is a complex topic - it may not work the same for all boards.